src/Symfony/Component/HttpFoundation/Tests/JsonResponseTest.php

```diff
@@ -43,7 +43,7 @@ public function testConstructorWithSimpleTypes()
         $this->assertSame('0', $response->getContent());
 
         $response = new JsonResponse(0.1);
-        $this->assertEquals('0.1', $response->getContent());
+        $this->assertEquals(0.1, $response->getContent());
```

Comment:
this change is wrong. We are precisely asserting the string content of the response here.

Reply:
But the string content can vary depending on the `ini` confs for floating point precision...
Here if the goal is to test the value being serialized correctly I think just evaluating the value should be the thing we want...

If we really care about `0.1` vs `0.10000000000000001` (I pasted the exact same value) we probably have to set some exact precision in the `.ini` file that PHPUnit is going to use...

One caveat of this test I could even catch, was `0.1` and `0.10` aren't also being treated the same...

Or it might be that I'm not correctly understanding the goal of this specific test though :sweat_smile:

Reply:
well, a PHP float is **not** a proper json content. This test is about asserting the fact that JsonResponse properly encodes various types in JSON.

Reply:
> This test is about asserting the fact that JsonResponse properly encodes various types in JSON.

Exactly, this what I thought as well :slightly_smiling_face: :+1: 
And that's why I proposed this change, because comparing string to string will have a lot of false-negatives (depending on how long the precision would be, 14 digits for Zend-engine default AFAIK).
But if we have one side of comparision as a normal number and the other side ( `$response->getContent()` ) as string, it'll force the Assert to convert both sides to the same type once (which will apply the local configs), and that way we'll no more have to worry about having a test `.ini` file and also we won't get this type of false-negatives :slightly_smiling_face: 

> well, a PHP float is not a proper json content

I'm afraid this wasn't clear for me :sweat_smile:  yes of course some floating-point 6bit data isn't what json sees as valid... but "0.100000001" is valid, no? :thinking: